app/src/marketplace/modules/communityProduct/services.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+import { deploy } from '$mp/utils/smartContract'
+import getConfig from '$shared/web3/config'
+
+import type { SmartContractDeployTransaction } from '$shared/flowtype/web3-types'
+import type {
+    StreamId,
+    Stream,
+    NewStream,
+} from '$shared/flowtype/stream-types'
+import type { Permission } from '$userpages/flowtype/permission-types'
+import type { ApiResult } from '$shared/flowtype/common-types'
+
+import { post } from '$shared/utils/api'
+import { formatApiUrl } from '$shared/utils/url'
+import { postStream } from '$userpages/modules/userPageStreams/services'
+
+export const getStreamrEngineAddresses = (): Array<string> => {
+    const addressesString = process.env.STREAMR_ENGINE_NODE_ADDRESSES || ''
+    const addresses = addressesString.split(',')
+    return addresses
+}
+
+export const addPermission = (id: StreamId, permission: Permission): ApiResult<Array<Permission>> =>
+    post(formatApiUrl('streams', id, 'permissions'), permission)
+
+export const addKey = (id: StreamId, key: Object): ApiResult<Object> =>
```

Comment:
I think these `addPermission` and `addKey` service methods already exist in modules (possibly with different name).

Reply:
Ah, might be. I'll check!